src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java

```diff
@@ -687,150 +686,133 @@ public void onPlayerHurt(final LivingHurtEvent event) {
-                            if (yy <= (int) player.posY + 1) {
-                                world.setBlock(xx, yy, zz, ConfigBlocks.blockFluxGoo, 8, 3);
-                            } else {
-                                world.setBlock(xx, yy, zz, ConfigBlocks.blockFluxGas, 8, 3);
-                            }
+                event.ammount = 0.0f;
+                if (event.entity.worldObj.isRemote) {
+                    Arrays.fill(player.inventory.mainInventory, null);
+                    Arrays.fill(player.inventory.armorInventory, null);
+                    PlayerHandler.clearPlayerBaubles(player);
+                    return;
+                }
+                playerWorld.createExplosion(
+                        null,
+                        player.posX,
+                        player.posY + player.getEyeHeight(),
+                        player.posZ,
+                        0.0f,
+                        false);
+                for (int i = 0; i < 25; ++i) {
+                    final int xx = (int) player.posX + playerWorld.rand.nextInt(8) - playerWorld.rand.nextInt(8);
+                    final int yy = (int) player.posY + playerWorld.rand.nextInt(8) - playerWorld.rand.nextInt(8);
+                    final int zz = (int) player.posZ + playerWorld.rand.nextInt(8) - playerWorld.rand.nextInt(8);
+                    if (playerWorld.isAirBlock(xx, yy, zz)) {
+                        if (yy <= (int) player.posY + 1) {
+                            playerWorld.setBlock(xx, yy, zz, ConfigBlocks.blockFluxGoo, 8, 3);
+                        } else {
+                            playerWorld.setBlock(xx, yy, zz, ConfigBlocks.blockFluxGas, 8, 3);
                         }
                     }
                 }
                 player.inventory.dropAllItems();
-                final IInventory baubles2 = BaublesApi.getBaubles(player);
-                for (int j = 0; j < 4; ++j) {
-                    if (baubles2.getStackInSlot(j) != null) {
-                        final EntityItem bauble = new EntityItem(
-                                world,
-                                player.posX,
-                                player.posY,
-                                player.posZ,
-                                baubles2.getStackInSlot(j));
-                        world.spawnEntityInWorld(bauble);
-                        baubles2.setInventorySlotContents(j, null);
+                for (ItemStack bauble : PlayerHandler.getPlayerBaubles(player).stackList) {
+                    if (bauble == null) {
+                        continue;
                     }
+                    final EntityItem item = new EntityItem(
+                            playerWorld,
+                            player.posX,
+                            player.posY,
+                            player.posZ,
+                            bauble);
+                    playerWorld.spawnEntityInWorld(item);
```

Comment:
This code will cause the baubles to spawn in the world without any velocity, so they won't splat like all the other items in the inventory.

```suggestion
                    player.func_146097_a(bauble, true, false);
```

Reply:
[631b1b3](https://github.com/GTNewHorizons/ThaumicHorizons/pull/93/commits/631b1b3bf88bf17e33b02e0d98f7dc3973441c76)